storagemarket/README.md

```diff
@@ -0,0 +1,345 @@
+```go
+func NewClient(
+	net network.StorageMarketNetwork,
+	bs blockstore.Blockstore,
+	dataTransfer datatransfer.Manager,
+	discovery *discovery.Local,
+	ds datastore.Batching,
+	scn storagemarket.StorageClientNode,
+) (*Client, error)
+```
+**Parameters**
+
+* `net network.StorageMarketNetwork` is a network abstraction for the storage market. To create it, use:
+    ```go
+    package network
+    func NewFromLibp2pHost(h host.Host) StorageMarketNetwork
+    ```
+* `bs blockstore.Blockstore` is an IPFS blockstore for storing and retrieving data for deals.
+     See [github.com/ipfs/go-ipfs-blockstore](github.com/ipfs/go-ipfs-blockstore).
+* `dataTransfer datatransfer.Manager` is an interface from [github.com/filecoin-project/go-data-transfer](https://github.com/filecoin-project/go-data-transfer)
+   There is more than one implementation, but one way to create a new datatransfer.Manager is:
+    ```go
+    package graphsyncimpl
+  
+    func NewGraphSyncDataTransfer(host host.Host, gs graphsync.GraphExchange, storedCounter *storedcounter.StoredCounter) datatransfer.Manager
+    ```
+   Also:
+   ```go
+    package datatransfer
+  
+    // NewDAGServiceDataTransfer returns a data transfer manager based on
+    // an IPLD DAGService
+    func NewDAGServiceDataTransfer(dag ipldformat.DAGService) datatransfer.Manager
+    ```
+    
+    Please see the [go-data-transfer repo](https://github.com/filecoin-project/go-data-transfer) for more information.
+    
+* `discovery *discovery.Local` implements the `PeerResolver` interface. To initialize a new discovery.Local:
+    ```go
+    func NewLocal(ds datastore.Batching) *Local
+    ```
+* `ds datastore.Batching` is a datastore for the deal's state machine. It is
+ typically the node's own datastore that implements the IPFS datastore.Batching interface.
+ See
+  [github.com/ipfs/go-datastore](https://github.com/ipfs/go-datastore).
+
+* `scn storagemarket.StorageClientNode` is the implementation of the [`StorageClientNode`](#StorageClientNode) API 
+that was written for your node.
+
+### StorageProvider
+To construct a new StorageProvider:
+```go
+func NewProvider(net network.StorageMarketNetwork, 
+                ds datastore.Batching, 
+                bs blockstore.Blockstore, 
+                fs filestore.FileStore, 
+                pieceStore piecestore.PieceStore, 
+                dataTransfer datatransfer.Manager, 
+                spn storagemarket.StorageProviderNode, 
+                minerAddress address.Address, 
+                rt abi.RegisteredProof, 
+                storedAsk StoredAsk, 
+                options ...StorageProviderOption,
+) (storagemarket.StorageProvider, error) {
+```
+
+**Parameters**
+* `net network.StorageMarketNetwork` is the same as for [StorageClientNode](#StorageClientNode)
+* `ds datastore.Batching` is the same as for [StorageClientNode](#StorageClientNode)
+* `bs blockstore.Blockstore` is the same blockstore as for [StorageClientNode](#StorageClientNode)
```

Comment:
These are the same interface, but not necessarily the same instance.  I seems a little bit ambiguous to me, but I'm not totally sure.